### PR TITLE
review from master

### PR DESCRIPTION
**Pull request by abhishek-rawat233**

### CODE REVIEW

File: app/models/hit_counter.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class HitCounter < ApplicationRecord
```
Comment (himanshumishra31) [inline app/models/hit_counter.rb]: Remove this

File: app/views/admin/categories.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      <th>Parent</th>
+      <th>Number of Products</th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @category.each do |category| %>
+      <tr>
+        <td><%= link_to category.name, categories_products_path(category.id) %></td>
+        <td><%= category.parent ? category.parent.name : 'no parent' %></td>
```
Comment (himanshumishra31) [inline app/views/admin/categories.html.erb]: Views helper

File: config/routes.rb [hunk 3]
```diff
@@ -1,31 +1,41 @@
...
-    get :line_items, to: "users#lineItems"
-  end
-  resources :products do
-    get :who_bought, on: :member
+    get :'/my-orders', to: "users#orders", as: 'orders'
+    get :'/my-items', to: "users#lineItems", as: 'line_items'
   end
 
+  resources :products, path: :books, as: :products
+    resources :products do
```
Comment (himanshumishra31) [inline config/routes.rb]: Indentation and DRY

File: lib/tasks/make_user_admin.rake [hunk 4]
```diff
@@ -0,0 +1,6 @@
+namespace :make_user_admin do
+  desc "this sets user's role to admin by checking mail"
+  task :set_user_to_admin, [:email] => :environment do |t, args|
+    User.where(email: args[:email]).update(role: 'admin')
```
Comment (himanshumishra31) [inline lib/tasks/make_user_admin.rake]: Single query

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,9 +1,39 @@
...
+        else
+          session[:inactive_time] = Time.now
+        end
+      end
+    end
+
+    def action_view_response_time
+      starting_time = Time.now
+      yield
+      time_taken = Time.now - starting_time
```
[the remark this answers is not in the document]
Comment (abhishek-rawat233) [reply app/controllers/application_controller.rb]: no such methods

**Pull request closed without merging**